src/main/webapp/app/course/manage/overview/course-management-statistics.component.ts

```diff
@@ -0,0 +1,108 @@
+import { Component, Input, OnChanges } from '@angular/core';
+import { Graphs } from 'app/entities/statistics.model';
+import { ChartDataSets, ChartType } from 'chart.js';
+import { Label } from 'ng2-charts';
+import { TranslateService } from '@ngx-translate/core';
+
+@Component({
+    selector: 'jhi-course-management-statistics',
+    templateUrl: './course-management-statistics.component.html',
+})
+export class CourseManagementStatisticsComponent implements OnChanges {
+    @Input()
+    courseId: number;
+
+    @Input()
+    amountOfStudentsInCourse: number;
+
+    @Input()
+    initialStats: number[];
+    initialStatsReceived = false;
+
+    graphType: Graphs = Graphs.ACTIVE_STUDENTS;
+
+    // Chart
+    chartName: string;
+
+    // Histogram-related properties
+    barChartOptions: any = {};
+    barChartType: ChartType = 'line';
+    amountOfStudents: string;
+    barChartLegend = false;
+
+    // Data
+    barChartLabels: Label[] = [];
+    chartData: ChartDataSets[] = [];
+    dataForSpanType: number[] = [];
+
+    constructor(private translateService: TranslateService) {}
+
+    ngOnChanges() {
+        this.amountOfStudents = this.translateService.instant('courseStatistics.amountOfStudents');
+
+        // Only use the pre-loaded stats once
+        if (this.initialStatsReceived || !this.initialStats || this.amountOfStudentsInCourse < 1) {
+            return;
+        }
+
+        this.initialStatsReceived = true;
```

Comment:
Can't you just do the initialization of the initial stats in `ngOnInit`? That should only be executed once  and you can simply remove the checks

Reply:
Hm, yeah this can probably change now that we rewrote how this works. Thanks